### PR TITLE
✨ GitLab: Documentation and cleaner errors

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Add documentation for GitLab support and show cleaner error messages (i.e., InconclusiveResult) if a check isn't supported. 

Also, some panics are fixed and logging improvements are made.

#### What is the current behavior?
Currently, we give 10/10 scores to GitLab in certain checks due to lack of data.

```release-note
* GitLab: For checks that aren't possible on GitLab, return Inconclusive Result
* GitHub: If no Workflows are detected, Dangerous-Workflows check returns Inconclusive instead of max score
```

### CODE REVIEW

File: checks/evaluation/dangerous_workflow.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ func DangerousWorkflow(name string, dl checker.DetailLogger,
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
+	if r.NumWorkflows == 0 {
```
Comment (azeemshaikh38) [inline checks/evaluation/dangerous_workflow.go:33]: Use `len(Workflows)`? Do we need this extra field in the struct?

File: checks/evaluation/permissions/permissions.go [hunk 2]
```diff
@@ -40,6 +40,10 @@ func TokenPermissions(name string, c *checker.CheckRequest, r *checker.TokenPerm
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
+	if r.NumTokens == 0 {
```
Comment (azeemshaikh38) [inline checks/evaluation/permissions/permissions.go:43]: Use `len(Tokens)` and remove this field.

File: checks/raw/dangerous_workflow.go [hunk 3]
```diff
@@ -102,6 +102,8 @@ var validateGitHubActionWorkflowPatterns fileparser.DoWhileTrueOnFileContent = f
 		return true, nil
 	}
 
+	pdata.NumWorkflows += 1
```
Comment (azeemshaikh38) [inline checks/raw/dangerous_workflow.go:105]: Remove.

File: checks/raw/permissions.go [hunk 4]
```diff
@@ -84,6 +84,8 @@ var validateGitHubActionTokenPermissions fileparser.DoWhileTrueOnFileContent = f
 		return true, nil
 	}
 
+	pdata.results.NumTokens += 1
```
Comment (azeemshaikh38) [inline checks/raw/permissions.go:87]: Remove.

**Review by azeemshaikh38 — APPROVED** [review]
Approved with comments.

File: checks/evaluation/dangerous_workflow.go (hunk 1 above)
Comment (raghavkaul) [reply checks/evaluation/dangerous_workflow.go:33]: This is intentional, `Workflows` doesn't have all workflows, just ones we've found with risky security patterns.

File: clients/gitlabrepo/branches.go [hunk 5]
```diff
@@ -73,7 +73,6 @@ func (handler *branchesHandler) setup() error {
 				handler.repourl.project, &gitlab.ListOptions{})
 			if err != nil && resp.StatusCode != 404 {
 				handler.errSetup = fmt.Errorf("request for external status checks failed with error %w", err)
-				return
```
[the remark this answers is not in the document]
Comment (raghavkaul) [reply clients/gitlabrepo/branches.go:76]: Yes, this was intentional - this API requires a more privileged API token. It typically returns 401 when the token isn't authorized. I can change this to ensure that `resp.StatusCode == 401` if that makes it clearer?

[the remark this answers is not in the document]
Comment (raghavkaul) [reply clients/gitlabrepo/branches.go:76]: What's the best way to silently pass a potential permission error here?

[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply clients/gitlabrepo/branches.go:76]: If we want `401` to not have an error, we should be setting `handler.errSetup` to be `nil`.

**Pull request merged**